src/d/lyt/d_lyt_pause.cpp

```diff
@@ -0,0 +1,1026 @@
+    }
+}
+
+void dLytPauseMgr_c::inRequest() {
+    mInRequest = true;
+    field_0x0836 = 1;
+}
+
+bool dLytPauseMgr_c::isStateWait() const {
+    return mStateMgr.getStateID()->isEqual(StateID_Wait);
+}
+
+void dLytPauseMgr_c::preDrawStage() {
+    if (field_0x0839 == true) {
+        field_0x0839 = false;
+        d2d::defaultSet();
+        if (mpTexture0x0810 != nullptr) {
+            mpTexture0x0810->free();
+            mpTexture0x0810 = nullptr;
+        }
+        mMain.drawDirectly();
+        mpTexture0x0810 = EGG::TextureBuffer::alloc(
+            mVideo::m_video->pRenderMode->fbWidth / 2, dGfx_c::getEFBHeight() / 2, GX_TF_RGBA8
+        );
+        mpTexture0x0810->capture(0, dGfx_c::getLetterboxAmount(), true, -1);
+        mMain.loadBgTextureUnused();
+    }
+}
+
+void dLytPauseMgr_c::postDrawStage() {
+        if (field_0x083A == true) {
+        field_0x083A = false;
+        d2d::defaultSet();
+        if (dLytControlGame_c::getInstance() != nullptr) {
+            dLytControlGame_c::getInstance()->fn_802D04F0();
+        }
+        mpBgTexture = EGG::TextureBuffer::alloc(
+            mVideo::m_video->pRenderMode->fbWidth / 2, dGfx_c::getEFBHeight() / 2, GX_TF_RGBA8
+        );
+        mpBgTexture->capture(0, dGfx_c::getLetterboxAmount(), true, -1);
+        mMain.loadBgTexture();
+    }
+}
+
+nw4r::lyt::Bounding *dLytPauseMgr_c::getArrowBounding(int idx) const {
+    return mMain.getArrow()->getArrowBounding(idx);
+}
+
+void dLytPauseMgr_c::setSelectedArrowBounding(int idx) {
+    mMain.getArrow()->setField_0x6B8(idx);
+}
+
+void dLytPauseMgr_c::setSelection(dLytPauseMgr_c::SelectionType_e type, u16 id, bool restricted) {
+    if (type != mPrevSelectionType || id != getCurrentSelectionId()) {
+        field_0x083B = true;
+    } else {
+        field_0x083B = false;
+    }
+    mCurrentSelectionType = type;
+    mCurrentSelectionId = id;
+    mCurrentSelectionIsRestricted = restricted;
+}
+
+bool dLytPauseMgr_c::checkSelectRing() {
+    dLytControlGame_c *lytControl = dLytControlGame_c::getInstance();
+    if (lytControl == nullptr) {
+        mCurrentDisp00Tab = TAB_ITEM;
+        return false;
+    }
+
+    if (lytControl->getCurrentPauseDisp() != DISP_00_INVENTORY) {
+        return false;
+    }
+
+    s32 tab = lytControl->getPauseDisp00Tab();
+
+    if (dPad::getDownTrigB()) {
+        mCurrentDisp00Tab = TAB_ITEM;
+        if (tab != TAB_ITEM && StoryflagManager::sInstance->getFlag(58)) {
+            return true;
+        } else {
+            return false;
+        }
+    } else if (dPad::getDownTrigMinus()) {
+        mCurrentDisp00Tab = TAB_POUCH;
+        if (tab != TAB_POUCH && StoryflagManager::sInstance->getFlag(30)) {
+            return true;
+        } else {
+            return false;
+        }
+    } else if (dPad::getDownTrigC()) {
+        mCurrentDisp00Tab = TAB_DOWSING;
+        if (tab != TAB_DOWSING && StoryflagManager::sInstance->getFlag(789)) {
```

Comment:
I am assuming these are flags for having them obtained, buta short comment describing it would be nice

Reply:
#242